src/pages/groupComparison/GroupComparisonUtils.tsx

```diff
@@ -1063,3 +1065,111 @@ const AlterationMenuHeader: React.FunctionComponent<{
         );
     }
 );
+
+// Benjamini-Hochberg procedure q-value, used to show significant association of mutations enriched in a group
+export const SIGNIFICANT_QVALUE_THRESHOLD = 0.05;
+
+export function getProteinChangeToMutationRowData(
+    tableData: Mutation[][],
+    mutationsGroupedByProteinChangeForGroup: (
+        groupIndex: number
+    ) => _.Dictionary<{
+        group: string;
+        data: any[];
+    }>,
+    profiledPatientsCounts: number[],
+    groups: ComparisonGroup[]
+): {
+    [proteinChange: string]: ComparisonMutationsRow;
+} {
+    let rowData = tableData.map(proteinChangeRow => {
+        const groupAMutatedCount: number =
+            getMutatedCountsByAttributeForGroup(
+                mutationsGroupedByProteinChangeForGroup,
+                0
+            )[proteinChangeRow[0].proteinChange] || 0;
+        const groupBMutatedCount: number =
+            getMutatedCountsByAttributeForGroup(
+                mutationsGroupedByProteinChangeForGroup,
+                1
+            )[proteinChangeRow[0].proteinChange] || 0;
+        const groupAMutatedPercentage: number =
+            (groupAMutatedCount / profiledPatientsCounts[0]) * 100;
+        const groupBMutatedPercentage: number =
+            (groupBMutatedCount / profiledPatientsCounts[1]) * 100;
+        const logRatio: number = Math.log2(
+            groupAMutatedPercentage / groupBMutatedPercentage
+        );
+        const pValue: number = getTwoTailedPValue(
+            groupAMutatedCount,
+            profiledPatientsCounts[0] - groupAMutatedCount,
+            groupBMutatedCount,
+            profiledPatientsCounts[1] - groupBMutatedCount
+        );
+        const enrichedGroup: string =
+            logRatio > 0
+                ? groups[0].nameWithOrdinal
+                : groups[1].nameWithOrdinal;
+
+        return {
+            proteinChange: proteinChangeRow[0].proteinChange,
+            groupAMutatedCount,
+            groupBMutatedCount,
+            groupAMutatedPercentage,
+            groupBMutatedPercentage,
+            logRatio,
+            pValue,
+            qValue: 0,
+            enrichedGroup,
+        };
+    });
+
+    rowData = _.sortBy(rowData, r => r.pValue);
+    const qValues = calculateQValues(_.map(rowData, d => d.pValue));
+    rowData.forEach((d, i) => {
+        d.qValue = qValues[i];
+    });
+
+    return _.keyBy(rowData, d => d.proteinChange);
+}
+
+export function getMutationCountsByAttributeForGroup(
+    getMutationsGroupedByAttributeForGroup: (
+        groupIndex: number
+    ) => _.Dictionary<{
+        group: string;
+        data: any[];
+    }>,
+    groupIndex: number
+): {
+    [attribute: string]: number;
+} {
+    const map: { [attribute: string]: number } = {};
+
+    _.forIn(getMutationsGroupedByAttributeForGroup(groupIndex), (v, k) => {
+        map[v.group] = v.data.length;
+    });
+    return map;
+}
+
+export function getMutatedCountsByAttributeForGroup(
```

Comment:
What's the difference between this one and `getMutationCountsByAttributeForGroup`? Would it be better if we unify them with an additional `uniqueCount` flag or a `uniqueBy` function as a third parameter?

Reply:
Mutated counts are the number of mutated patients (those with mutations) and mutation counts are number of mutations. I can make it one function with an additional parameter that's a boolean for mutation/mutated counts, and depending on that we do the appropriate action